### PR TITLE
SpeciesInfo tests

### LINKED ISSUE #3689: The Mimikyu animation doesn't have a family toggle.
### Description

In src/data/pokemon_graphics/front_pic_anims.h, PLACEHOLDER_ANIM_SINGLE_FRAME(MimikyuDisguised) & PLACEHOLDER_ANIM_SINGLE_FRAME(MimikyuBusted) are in P_FAMILY_TOGEDEMARU.

### Version

upcoming (Edge)

### Upcoming Version

ec9cc9a

### Discord contact info

Iván#7099

### PR DESCRIPTION
**Pull request by mrgriffin**
Adds tests that verify appropriate invariants hold between the members of `gSpeciesInfo`:
- All forms of a species have the same `.formSpeciesIdTable`.
- All form changes' `.targetSpecies` are to a species ID in `.formSpeciesIdTable`.

Also corrects some small issues I noticed while writing and fixing these tests.

Closes #3689

### CODE REVIEW

File: src/data/pokemon/species_info.h [hunk 1]
```diff
@@ -48948,6 +48949,7 @@ const struct SpeciesInfo gSpeciesInfo[] =
         .categoryName = _("Drummer"),                                           \
         .footprint = gMonFootprint_Rillaboom,                                   \
         LEARNSETS(Rillaboom),                                                   \
+        .formSpeciesIdTable = sRillaboomFormSpeciesIdTable,                     \
```
Comment (Bassoonian) [inline src/data/pokemon/species_info.h:48952]: I understand that they were missing from the base forms, but shouldn't they be in the Gigantamax forms also? The Mega Evolutions certainly seem to have the tables assigned to all forms, so we should probably be consistent (unless there's a reason not to be?)

Comment (AsparagusEduardo) [reply src/data/pokemon/species_info.h:48952]: Yes, all forms of a species should have the table assigned.

Comment (mrgriffin) [reply src/data/pokemon/species_info.h:48952]: It's not at all clear here on GitHub, but this is a change to the `RILLABOOM_MISC_INFO` macro which is used by both `SPECIES_RILLABOOM` and `SPECIES_RILLABOOM_GIGANTAMAX`. So it should be applied to both the base form and the Gigantamax forms.

Comment (AsparagusEduardo) [reply src/data/pokemon/species_info.h:48952]: Right, sorry, just woke up xD

Comment (Bassoonian) [reply src/data/pokemon/species_info.h:48952]: Apologies, my bad :) all good to go then!

Comment (mrgriffin) [reply src/data/pokemon/species_info.h:48952]: np! It's really difficult to see on GitHub!

**Pull request merged**